libglusterfs/src/syncop.c

```diff
@@ -474,7 +469,16 @@ synctask_create(struct syncenv *env, size_t stacksize, synctask_fn_t fn,
     if (destroymode)
         return NULL;
 
-    newtask = GF_CALLOC(1, sizeof(*newtask), gf_common_mt_synctask);
+    if (stacksize <= 0) {
+        newtask = GF_CALLOC(1, sizeof(struct synctask) + env->stacksize,
```

Comment:
I would try to use `GF_MALLOC()`. The stack is of significant size and there's no reason to clear it. Without initializing it it's even possible that some portions of the stack won't be mapped until needed, reducing the physical memory utilization.

Reply:
I think the initialization is fine. At least I haven't seen any issue.

A possible alternative for other cases (specially useful when most of the fields are initialized to 0) is this:

```c
ptr = malloc(sizeof(type) + extra);
memset(ptr, 0, sizeof(type));
```